dot/sync/chain_sync_test.go

```diff
@@ -827,3 +827,189 @@ func TestChainSync_determineSyncPeers(t *testing.T) {
 	require.Equal(t, 1, len(peers))
 	require.Equal(t, []peer.ID{testPeerB}, peers)
 }
+
+func TestChainSync_highestBlock(t *testing.T) {
+	type input struct {
+		peerState map[peer.ID]*peerState
+	}
+	type output struct {
+		highestBlock int64
+		err          error
+	}
+	type test struct {
+		name string
+		in   input
+		out  output
+	}
+	tests := []test{
+		{
+			name: "when has an empty map should return 0, errNoPeers",
+			in: input{
+				peerState: map[peer.ID]*peerState{},
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNoPeers,
+			},
+		},
+		{
+			name: "when has a nil map should return 0, errNoPeers",
+			in: input{
+				peerState: map[peer.ID]*peerState{},
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNoPeers,
+			},
+		},
+		{
+			name: "when has only one peer with number 90 should return 90, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest": {
+						who:    "idtest#1",
+						hash:   common.BytesToHash([]byte("hash#1")),
+						number: big.NewInt(90),
+					},
+				},
+			},
+			out: output{
+				highestBlock: int64(90),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has only one peer with number nil should return 0, errNilBlockData",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest": {
+						who:    "idtest#1",
+						hash:   common.BytesToHash([]byte("hash#1")),
+						number: nil,
+					},
+				},
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNilBlockData,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number 90 and p2.number 190 should return 190, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {
+						who:    "idtest#1",
+						hash:   common.BytesToHash([]byte("hash#1")),
+						number: big.NewInt(90),
+					},
+					"idtest#2": {
+						who:    "idtest#2",
+						hash:   common.BytesToHash([]byte("hash#2")),
+						number: big.NewInt(190),
+					},
+				},
+			},
+			out: output{
+				highestBlock: int64(190),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number 190 and p2.number 90 should return 190, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {
+						who:    "idtest#1",
+						hash:   common.BytesToHash([]byte("hash#1")),
+						number: big.NewInt(190),
+					},
+					"idtest#2": {
+						who:    "idtest#2",
+						hash:   common.BytesToHash([]byte("hash#2")),
+						number: big.NewInt(90),
+					},
+				},
+			},
+			out: output{
+				highestBlock: int64(190),
+				err:          nil,
```

Comment:
remove setting empty fields (apply to other test cases as well)
```suggestion
```

Reply:
If a new developer face those tests without fields, I believe he/she could be stunned in a first moment... Although this cause more code, I think it is more clear (as a developer) when all fields filled out. Of course, there is cases that isn't good to have that but in this case I think is useful.

What you think? I can change it but I would like to know your opinion before.

Reply:
Well I have to admit I am big fan of explicitness. 
On the other hand, settings all empty fields explicitly to their empty value does add a lot of code, especially for more complex test cases or a lot of test cases, making it quite unreadable.

Some like [Uber also advise to leave empty fields empty](https://github.com/uber-go/guide/blob/master/style.md#omit-zero-value-fields-in-structs).

Although again, in this particular case, up to you it doesn't matter much 😉

Reply:
I've never seen this guide, cool! I can use that to defend some arguments in the future 😆 

So, in this case I would prefer keep it. These zero values could help to understand these tests, because names are written in this template: "when {{input}} should return {{output}}" and the uber guide says:

```
Include zero values where field names provide meaningful context. For example, test cases in Test Tables can benefit from names of fields even when they are zero-valued.

tests := []struct{
  give string
  want int
}{
  {give: "0", want: 0},
  // ...
}
```

I think I can remove some fields from `peerState` like: `who` and `hash` because they aren't being used in this test.

Just to finish, could we close this conversation if I do these things(?):

* Remove `who` and `hash` from `peerState` (they aren't provide any meaningful)
* Keep others meaningful zero values for tests.

Reply:
Yes that makes sense! Good catch the Uber guide bite me back 😆  I'll resolve it and let it up to you 😉